core/src/main/scala/org/apache/pekko/persistence/jdbc/journal/dao/BaseJournalDaoWithReadMessages.scala

```diff
@@ -38,13 +38,28 @@ trait BaseJournalDaoWithReadMessages extends JournalDaoWithReadMessages {
       toSequenceNr: Long,
       batchSize: Int,
       refreshInterval: Option[(FiniteDuration, Scheduler)]): Source[Try[(PersistentRepr, Long)], NotUsed] = {
+    internalBatchStream(persistenceId, fromSequenceNr, toSequenceNr, batchSize, refreshInterval).mapConcat(identity)
+  }
 
+  /**
+   * separate this method for unit tests.
+   */
+  @InternalApi
+  private[dao] def internalBatchStream(persistenceId: String,
+      fromSequenceNr: Long,
+      toSequenceNr: Long,
+      batchSize: Int,
+      refreshInterval: Option[(FiniteDuration, Scheduler)]) = {
```

Comment:
should add some check for `fromSequenceNr` and `batchSize`

Reply:
and add checking for where `fromSequenceNr > toSequenceNr` too

Reply:
> should add some check for fromSequenceNr and batchSize

what does that mean? batch size didn't allow `zero` value, and the `fromSequenceNr` was generate from acotr it self.

> and add checking for where fromSequenceNr > toSequenceNr too

`fromSequenceNr > toSequenceNr` is enforced down below.

Reply:
it is generally good practice to do input validation to catch failures early and make constraints explicit